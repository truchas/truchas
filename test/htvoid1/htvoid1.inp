HTVOID1 -- Heat transfer in flow with void, Test 1

This exercises the treatment of void within the diffusion solver's
heat transfer model.  The focus here is on the dynamic domain for
the heat equation and on the advected heat source.  The domain is
a 1D flow channel with a 3x3 cell cross section.  There is a plug
of fluid near the left end, with void on both sides of the plug.
The plug is initially at rest but accelerates down the channel
due to the gravity which is directed to the right.  The simulation
terminates before the fluid has reached the right end.  The mesh
nodes are randomly perturbed (10% of the edge length).  There is
an initial temperature gradient in the fluid, and no-heat-flux BC
imposed everywhere.  With a small conductivity, the initial temperature
profile should be carried down the channel with the fluid without
large changes.

The flow itself is extremely sensitive to the various flow parameters
(but completely independent of the heat transfer).  Small changes to
things -- initial step size, for example -- will lead to different
flow solutions with bits of fluid left behind or not.  The flow
solution is largely correct, but the differences in which cells are
void will result in heat solutions that are not comparable.

&MESH
  mesh_file = 'channel-3.gen'
  mesh_file_format = 'ExodusII'
/

&OUTPUTS
  Output_T  = 0.0, 1.0
  Output_Dt = 1.0 !.05
/

&PHYSICS
  fluid_flow = .true.
  inviscid   = .true.
  body_force = 6.0, 0.0, 0.0
  heat_transport = .true.
/

&DIFFUSION_SOLVER
  stepping_method    = 'Non-adaptive BDF1'
  residual_atol      = 0.0
  residual_rtol      = 1.0e-6
  max_nlk_itr        = 10
  nlk_preconditioner = 'hypre_amg'
  verbose_stepping   = .true.
/

&NUMERICS
  dt_init = 0.01
  dt_grow = 1.05
  projection_linear_solution = 'projection'
  body_force_face_method = .true.
  volume_track_interfaces    = .true.
  volume_track_Brents_Method = .true.
  volume_track_subcycles     = 2
  volume_track_iter_Tol      = 1.0e-12
  cutvof                     = 1.0e-8
  courant_number             = 0.4
  discrete_ops_type          = 'ortho' 
/

&LINEAR_SOLVER
  name                   = 'projection'
  method                 = 'gmres'
  convergence_criterion  = 1.0e-12
  preconditioning_method = 'ssor'
  relaxation_parameter   = 1.4
  preconditioning_steps  = 4
  maximum_iterations     = 500
  output_mode            = 'summary'
/

! Fails in parallel; see artf19997.
!&INTERFACES
!  vof_method = 'divide'
!/

Fluid plug at left end of the channel
&BODY
  surface_name    = 'box'
  length = 3.0, 2.0, 2.0
  translation_pt = -1.5, 0.0, 0.0
  material_number = 1
  velocity        = 0.5, 0.0, 0.0
  temperature_function = 'init-temp'
/

T = 6 + (x + y + z)/sqrt(3)
&FUNCTION
  name = 'init-temp'
  type = 'polynomial'
  poly_coefficients = 6, 3*0.577350269
  poly_exponents(:,1) = 0, 0, 0
  poly_exponents(:,2) = 1, 0, 0
  poly_exponents(:,3) = 0, 1, 0
  poly_exponents(:,4) = 0, 0, 1
/

Remainder of the channel is void
&BODY
  surface_name    = 'background'
  mesh_material_number = 1
  material_number = 2
  temperature     = 1.0
  velocity = 3*0.0
/

&DS_BOUNDARY_CONDITION
  name = 'whole boundary'
  variable = 'temperature'
  condition = 'flux'
  face_set_ids = 1, 2, 3, 4, 5, 6
  data_constant = 0.0
/

&MATERIAL
  Material_Name   = 'fluid'
  Material_Number = 1
  priority        = 1
  Density         = 1.0
/

&PHASE
  name = 'fluid'
  property_name(1) = 'density',       property_constant(1) = 1.0
  property_name(2) = 'specific heat', property_constant(2) = 1.0
  property_name(3) = 'conductivity',  property_constant(3) = 1.0e-3
/

&MATERIAL_SYSTEM
  name = 'fluid'
  phases = 'fluid'
/

&MATERIAL
  Material_Name = 'void'
  Material_Number = 2
  Material_Feature = 'background'
  priority = 2
  Density = 0.0
/
