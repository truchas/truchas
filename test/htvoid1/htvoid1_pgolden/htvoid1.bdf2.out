HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 5.498E-04 (initial)
  HC error: ||F||_2 = 1.327E-08, scaled = 2.414E+01
    1: error= 2.41433E+01
  HC error: ||F||_2 = 2.647E-11, scaled = 4.815E-02
    2: error= 4.81477E-02
  NLK BCE solve succeeded:   2 iterations, error= 4.81477E-02
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 3.500E-05 (initial)
  HC error: ||F||_2 = 1.402E-08, scaled = 4.005E+02
    1: error= 4.00464E+02
  HC error: ||F||_2 = 2.826E-11, scaled = 8.075E-01
    2: error= 8.07471E-01
  NLK BCE solve succeeded:   2 iterations, error= 8.07471E-01
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 5.312E-04 (initial)
  HC error: ||F||_2 = 2.508E-10, scaled = 4.722E-01
    1: error= 4.72210E-01
  HC error: ||F||_2 = 7.984E-13, scaled = 1.503E-03
    2: error= 1.50306E-03
  NLK BCE solve succeeded:   2 iterations, error= 1.50306E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 5.512E-04 (initial)
  HC error: ||F||_2 = 2.763E-10, scaled = 5.012E-01
    1: error= 5.01189E-01
  HC error: ||F||_2 = 8.969E-13, scaled = 1.627E-03
    2: error= 1.62706E-03
  NLK BCE solve succeeded:   2 iterations, error= 1.62706E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 5.773E-04 (initial)
  HC error: ||F||_2 = 2.995E-10, scaled = 5.187E-01
    1: error= 5.18674E-01
  HC error: ||F||_2 = 9.970E-13, scaled = 1.727E-03
    2: error= 1.72688E-03
  NLK BCE solve succeeded:   2 iterations, error= 1.72688E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 5.958E-04 (initial)
  HC error: ||F||_2 = 3.294E-10, scaled = 5.530E-01
    1: error= 5.52978E-01
  HC error: ||F||_2 = 1.140E-12, scaled = 1.913E-03
    2: error= 1.91332E-03
  NLK BCE solve succeeded:   2 iterations, error= 1.91332E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 6.363E-04 (initial)
  HC error: ||F||_2 = 3.681E-10, scaled = 5.786E-01
    1: error= 5.78581E-01
  HC error: ||F||_2 = 1.269E-12, scaled = 1.994E-03
    2: error= 1.99388E-03
  NLK BCE solve succeeded:   2 iterations, error= 1.99388E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 7.141E-04 (initial)
  HC error: ||F||_2 = 4.162E-10, scaled = 5.828E-01
    1: error= 5.82795E-01
  HC error: ||F||_2 = 1.109E-12, scaled = 1.553E-03
    2: error= 1.55311E-03
  NLK BCE solve succeeded:   2 iterations, error= 1.55311E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 7.475E-04 (initial)
  HC error: ||F||_2 = 4.541E-10, scaled = 6.074E-01
    1: error= 6.07449E-01
  HC error: ||F||_2 = 1.311E-12, scaled = 1.754E-03
    2: error= 1.75423E-03
  NLK BCE solve succeeded:   2 iterations, error= 1.75423E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 7.441E-04 (initial)
  HC error: ||F||_2 = 4.771E-10, scaled = 6.413E-01
    1: error= 6.41255E-01
  HC error: ||F||_2 = 1.412E-12, scaled = 1.898E-03
    2: error= 1.89832E-03
  NLK BCE solve succeeded:   2 iterations, error= 1.89832E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 7.533E-04 (initial)
  HC error: ||F||_2 = 5.173E-10, scaled = 6.866E-01
    1: error= 6.86622E-01
  HC error: ||F||_2 = 1.504E-12, scaled = 1.996E-03
    2: error= 1.99647E-03
  NLK BCE solve succeeded:   2 iterations, error= 1.99647E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 7.726E-04 (initial)
  HC error: ||F||_2 = 5.657E-10, scaled = 7.322E-01
    1: error= 7.32213E-01
  HC error: ||F||_2 = 1.609E-12, scaled = 2.083E-03
    2: error= 2.08298E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.08298E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 8.016E-04 (initial)
  HC error: ||F||_2 = 6.205E-10, scaled = 7.740E-01
    1: error= 7.74031E-01
  HC error: ||F||_2 = 1.744E-12, scaled = 2.175E-03
    2: error= 2.17509E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.17509E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 8.293E-04 (initial)
  HC error: ||F||_2 = 6.771E-10, scaled = 8.165E-01
    1: error= 8.16523E-01
  HC error: ||F||_2 = 1.882E-12, scaled = 2.269E-03
    2: error= 2.26896E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.26896E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 8.593E-04 (initial)
  HC error: ||F||_2 = 7.384E-10, scaled = 8.593E-01
    1: error= 8.59322E-01
  HC error: ||F||_2 = 2.025E-12, scaled = 2.357E-03
    2: error= 2.35679E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.35679E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 8.901E-04 (initial)
  HC error: ||F||_2 = 8.027E-10, scaled = 9.018E-01
    1: error= 9.01783E-01
  HC error: ||F||_2 = 2.164E-12, scaled = 2.431E-03
    2: error= 2.43138E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.43138E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 9.219E-04 (initial)
  HC error: ||F||_2 = 8.731E-10, scaled = 9.471E-01
    1: error= 9.47102E-01
  HC error: ||F||_2 = 2.301E-12, scaled = 2.496E-03
    2: error= 2.49582E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.49582E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 9.546E-04 (initial)
  HC error: ||F||_2 = 9.446E-10, scaled = 9.895E-01
    1: error= 9.89499E-01
  HC error: ||F||_2 = 2.421E-12, scaled = 2.537E-03
    2: error= 2.53662E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.53662E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 9.884E-04 (initial)
  HC error: ||F||_2 = 1.018E-09, scaled = 1.030E+00
    1: error= 1.02977E+00
  HC error: ||F||_2 = 2.515E-12, scaled = 2.545E-03
    2: error= 2.54466E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.54466E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 1.024E-03 (initial)
  HC error: ||F||_2 = 1.096E-09, scaled = 1.071E+00
    1: error= 1.07115E+00
  HC error: ||F||_2 = 2.668E-12, scaled = 2.607E-03
    2: error= 2.60692E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.60692E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 1.192E-03 (initial)
  HC error: ||F||_2 = 1.177E-09, scaled = 9.879E-01
    1: error= 9.87908E-01
  HC error: ||F||_2 = 3.084E-12, scaled = 2.588E-03
    2: error= 2.58771E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.58771E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 2.023E-03 (initial)
  HC error: ||F||_2 = 2.599E-09, scaled = 1.285E+00
    1: error= 1.28455E+00
  HC error: ||F||_2 = 5.947E-12, scaled = 2.939E-03
    2: error= 2.93916E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.93916E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 1.414E-03 (initial)
  HC error: ||F||_2 = 5.035E-09, scaled = 3.562E+00
    1: error= 3.56163E+00
  HC error: ||F||_2 = 1.442E-11, scaled = 1.020E-02
    2: error= 1.01967E-02
  NLK BCE solve succeeded:   2 iterations, error= 1.01967E-02
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 1.471E-03 (initial)
  HC error: ||F||_2 = 2.319E-09, scaled = 1.576E+00
    1: error= 1.57613E+00
  HC error: ||F||_2 = 1.046E-11, scaled = 7.108E-03
    2: error= 7.10752E-03
  NLK BCE solve succeeded:   2 iterations, error= 7.10752E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 1.539E-03 (initial)
  HC error: ||F||_2 = 2.549E-09, scaled = 1.656E+00
    1: error= 1.65619E+00
  HC error: ||F||_2 = 1.146E-11, scaled = 7.450E-03
    2: error= 7.44998E-03
  NLK BCE solve succeeded:   2 iterations, error= 7.44998E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 1.832E-03 (initial)
  HC error: ||F||_2 = 3.210E-09, scaled = 1.752E+00
    1: error= 1.75229E+00
  HC error: ||F||_2 = 1.464E-11, scaled = 7.991E-03
    2: error= 7.99072E-03
  NLK BCE solve succeeded:   2 iterations, error= 7.99072E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 2.523E-03 (initial)
  HC error: ||F||_2 = 5.345E-09, scaled = 2.118E+00
    1: error= 2.11849E+00
  HC error: ||F||_2 = 2.096E-11, scaled = 8.307E-03
    2: error= 8.30719E-03
  NLK BCE solve succeeded:   2 iterations, error= 8.30719E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 2.335E-03 (initial)
  HC error: ||F||_2 = 3.853E-09, scaled = 1.650E+00
    1: error= 1.65039E+00
  HC error: ||F||_2 = 2.028E-11, scaled = 8.685E-03
    2: error= 8.68537E-03
  NLK BCE solve succeeded:   2 iterations, error= 8.68537E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 2.569E-03 (initial)
  HC error: ||F||_2 = 3.895E-09, scaled = 1.516E+00
    1: error= 1.51636E+00
  HC error: ||F||_2 = 1.114E-11, scaled = 4.338E-03
    2: error= 4.33757E-03
  NLK BCE solve succeeded:   2 iterations, error= 4.33757E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 3.246E-03 (initial)
  HC error: ||F||_2 = 5.655E-09, scaled = 1.742E+00
    1: error= 1.74244E+00
  HC error: ||F||_2 = 2.574E-11, scaled = 7.931E-03
    2: error= 7.93138E-03
  NLK BCE solve succeeded:   2 iterations, error= 7.93138E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 2.718E-03 (initial)
  HC error: ||F||_2 = 3.467E-09, scaled = 1.276E+00
    1: error= 1.27552E+00
  HC error: ||F||_2 = 1.490E-11, scaled = 5.484E-03
    2: error= 5.48371E-03
  NLK BCE solve succeeded:   2 iterations, error= 5.48371E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 3.022E-03 (initial)
  HC error: ||F||_2 = 3.720E-09, scaled = 1.231E+00
    1: error= 1.23105E+00
  HC error: ||F||_2 = 6.454E-12, scaled = 2.135E-03
    2: error= 2.13547E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.13547E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 3.632E-03 (initial)
  HC error: ||F||_2 = 3.826E-09, scaled = 1.053E+00
    1: error= 1.05333E+00
  HC error: ||F||_2 = 6.848E-12, scaled = 1.885E-03
    2: error= 1.88519E-03
  NLK BCE solve succeeded:   2 iterations, error= 1.88519E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 3.179E-03 (initial)
  HC error: ||F||_2 = 3.848E-09, scaled = 1.210E+00
    1: error= 1.21011E+00
  HC error: ||F||_2 = 7.966E-12, scaled = 2.505E-03
    2: error= 2.50545E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.50545E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 3.514E-03 (initial)
  HC error: ||F||_2 = 1.495E-08, scaled = 4.255E+00
    1: error= 4.25541E+00
  HC error: ||F||_2 = 1.144E-10, scaled = 3.257E-02
    2: error= 3.25730E-02
  NLK BCE solve succeeded:   2 iterations, error= 3.25730E-02
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 3.710E-03 (initial)
  HC error: ||F||_2 = 2.821E-08, scaled = 7.604E+00
    1: error= 7.60449E+00
  HC error: ||F||_2 = 1.592E-10, scaled = 4.292E-02
    2: error= 4.29227E-02
  NLK BCE solve succeeded:   2 iterations, error= 4.29227E-02
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 3.179E-03 (initial)
  HC error: ||F||_2 = 4.077E-09, scaled = 1.283E+00
    1: error= 1.28256E+00
  HC error: ||F||_2 = 7.693E-12, scaled = 2.420E-03
    2: error= 2.41998E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.41998E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 3.864E-03 (initial)
  HC error: ||F||_2 = 4.807E-09, scaled = 1.244E+00
    1: error= 1.24406E+00
  HC error: ||F||_2 = 7.412E-12, scaled = 1.918E-03
    2: error= 1.91833E-03
  NLK BCE solve succeeded:   2 iterations, error= 1.91833E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 3.410E-03 (initial)
  HC error: ||F||_2 = 4.296E-09, scaled = 1.260E+00
    1: error= 1.25971E+00
  HC error: ||F||_2 = 9.816E-12, scaled = 2.878E-03
    2: error= 2.87822E-03
  NLK BCE solve succeeded:   2 iterations, error= 2.87822E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 3.428E-03 (initial)
  HC error: ||F||_2 = 4.306E-09, scaled = 1.256E+00
    1: error= 1.25617E+00
  HC error: ||F||_2 = 1.562E-11, scaled = 4.558E-03
    2: error= 4.55757E-03
  NLK BCE solve succeeded:   2 iterations, error= 4.55757E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 3.754E-03 (initial)
  HC error: ||F||_2 = 5.094E-09, scaled = 1.357E+00
    1: error= 1.35704E+00
  HC error: ||F||_2 = 1.950E-11, scaled = 5.195E-03
    2: error= 5.19500E-03
  NLK BCE solve succeeded:   2 iterations, error= 5.19500E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 3.283E-03 (initial)
  HC error: ||F||_2 = 3.861E-09, scaled = 1.176E+00
    1: error= 1.17627E+00
  HC error: ||F||_2 = 1.427E-11, scaled = 4.348E-03
    2: error= 4.34763E-03
  NLK BCE solve succeeded:   2 iterations, error= 4.34763E-03
HEAT TRANSFER FUNCTION NORMS
  HC error: ||F||_2 = 3.354E-03 (initial)
  HC error: ||F||_2 = 4.175E-09, scaled = 1.245E+00
    1: error= 1.24456E+00
  HC error: ||F||_2 = 2.088E-11, scaled = 6.226E-03
    2: error= 6.22569E-03
  NLK BCE solve succeeded:   2 iterations, error= 6.22569E-03
